airbyte-integrations/connectors/source-twilio/source_twilio/streams.py

```diff
@@ -90,45 +100,53 @@ def custom_transform_function(original_value: Any, field_schema: Mapping[str, An
+        self._cursor_value = None
 
     @property
     @abstractmethod
     def incremental_filter_field(self) -> str:
         """
         return: date filter query parameter name
         """
+    
+    @property
+    def state(self) -> Mapping[str, Any]:
+        if self._cursor_value:
+            return  self._cursor_value
+        else:
+            return  self._start_date
+
+    @state.setter
+    def state(self, value: Mapping[str, Any]):
+        self._cursor_value = value
 
-    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
-        """
-        Return the latest state by comparing the cursor value in the latest record with the stream's most recent state object
-        and returning an updated state object.
-        """
-        latest_benchmark = pendulum.parse(latest_record[self.cursor_field], strict=False).strftime(self.time_filter_template)
-        if current_stream_state.get(self.cursor_field):
-            return {self.cursor_field: max(latest_benchmark, current_stream_state[self.cursor_field])}
-        return {self.cursor_field: latest_benchmark}
 
     def request_params(self, stream_state: Mapping[str, Any], **kwargs) -> MutableMapping[str, Any]:
         params = super().request_params(stream_state=stream_state, **kwargs)
-        start_date = stream_state.get(self.cursor_field) or self._start_date
+        start_date = stream_state or self.state
         if start_date:
-            params.update({self.incremental_filter_field: pendulum.parse(start_date, strict=False).strftime(self.time_filter_template)})
+            params.update({self.incremental_filter_field: (pendulum.parse(start_date, strict=False)-timedelta(minutes=int(self._lookback))).strftime(self.time_filter_template)})
         return params
 
-    def read_records(self, stream_state: Mapping[str, Any] = None, **kwargs):
-        stream_state = stream_state or {}
-        records = super().read_records(stream_state=stream_state, **kwargs)
-        for record in records:
-            record[self.cursor_field] = pendulum.parse(record[self.cursor_field], strict=False).strftime(self.time_filter_template)
+    def read_records(self, *args, **kwargs) -> Iterable[Mapping[str, Any]]:
+        for record in super().read_records(*args, **kwargs):
+            if self._cursor_value:
+                record[self.cursor_field] = pendulum.parse(record[self.cursor_field], strict=False).strftime(self.time_filter_template)
+                self._cursor_value = datetime.strftime(max(datetime.strptime(self._cursor_value, self.time_filter_template), datetime.strptime(record[self.cursor_field],self.time_filter_template)),self.time_filter_template)
+            else:
+                record[self.cursor_field] = pendulum.parse(record[self.cursor_field], strict=False).strftime(self.time_filter_template)
+                self._cursor_value = record[self.cursor_field] or self.state
```

Comment:
I think the following logic should be deferred to the state setter as I suggested above:
```suggestion
        for record in super().read_records(*args, **kwargs):
            self._cursor_value = pendulum.parse(record[self.cursor_field], strict=False)
```
As you are comparing dates I don't think you should store string with `strftime`